src/snowflake/snowpark/file_operation.py

```diff
@@ -207,3 +207,52 @@ def get(
         # connector raises IndexError when no file is downloaded from python connector.
         except IndexError:
             return []
+
+    def put_stream(
```

Comment:
I just took a look at the counterpart in scala and actually there are two subtle differences: https://github.com/snowflakedb/thundersnow/blob/4731ff818927154f66d90fec0f2eaf91b1a17090/src/main/scala/com/snowflake/snowpark/FileOperation.scala#L146-L160
1. Should it be put_stream or upload_stream? (I slightly prefer put_stream)
2. Should we support a full stage path with `stage_location`, or support the separate `dest_file_name`? The user can specify `stage_location='@/xxx/yyy.csv'` and doesn't need `dest_file_name`. I feel it looks neater (also aligns with scala). 

@sfc-gh-sfan @sfc-gh-yixie thoughts?

Reply:
1. I also prefer `put_stream`. 
2. Yeah combining into one parameter seems more concise.

Reply:
I agree with `put_stream` and one parameter.